metadata-integration/java/openlineage-converter/src/main/java/io/datahubproject/openlineage/converter/OpenLineageToDataHub.java

```diff
@@ -632,13 +632,130 @@ private static void convertJobToDataJob(
     DataJobInfo dji = new DataJobInfo();
 
     log.debug("Datahub Config: {}", datahubConf);
-    if (job.getName().contains(".")) {
-
-      String jobName = job.getName().substring(job.getName().indexOf(".") + 1);
-      dji.setName(jobName);
-    } else {
-      dji.setName(job.getName());
+    
+    // Check if we have a MERGE INTO command
+    boolean isMergeIntoCommand = job.getName().contains("execute_merge_into_command_edge");
+    String tableName = null;
+    
+    // If this is a MERGE INTO command and enhanced extraction is enabled, try to extract the target table name
```

Comment:
Can we move this as well into a separate method to make this method a bit more digestable